### PR TITLE
Nft bulk transfer

### PR DESCRIPTION
**Pull request by ytx1991**
Add the bulk NFT transfer RPC API

### Purpose:
[#
](https://github.com/orgs/Chia-Network/projects/13?pane=issue&itemId=16507461)

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -186,7 +186,8 @@ def get_routes(self) -> Dict[str, Endpoint]:
             "/nft_add_uri": self.nft_add_uri,
             "/nft_calculate_royalties": self.nft_calculate_royalties,
             "/nft_mint_bulk": self.nft_mint_bulk,
-            "/nft_set_did_bulk": self.nft_set_bulk_nft_did,
+            "/nft_set_did_bulk": self.nft_set_did_bulk,
```
Comment (trepca) [inline chia/rpc/wallet_rpc_api.py:189]: nice

File: tests/wallet/nft_wallet/test_nft_wallet.py [hunk 2]
```diff
@@ -1308,6 +1308,132 @@ async def test_nft_bulk_set_did(self_hostname: str, two_wallet_nodes: Any, trust
...
+    ]
+    resp = await api_0.nft_transfer_bulk(dict(target_address=address, nft_coin_list=nft_coin_list))
+    sb = await make_new_block_with(resp, full_node_api, ph)
+    # ensure hints are generated
+    assert compute_memos(sb)
+    await time_out_assert(30, get_wallet_number, 2, wallet_node_1.wallet_state_manager)
+    coins_response = await wait_rpc_state_condition(
+        30, api_1.nft_get_nfts, [{"wallet_id": 2}], lambda x: len(x["nft_list"]) == 2
+    )
+    coins = coins_response["nft_list"]
```
Comment (trepca) [inline tests/wallet/nft_wallet/test_nft_wallet.py:1434]: you should check that correct coins are transferred over and also that DID is reset

**Comment by wallentx** [comment]
Is there an upper limit to how many can be transferred at once?

**Comment by ytx1991** [comment]
> Is there an upper limit to how many can be transferred at once?

Yes, it is controlled by MAX_NFT_CHUNK_SIZE = 25.

File: chia/rpc/wallet_rpc_api.py [hunk 3]
```diff
@@ -2215,7 +2216,7 @@ async def nft_set_bulk_nft_did(self, request):
         nft_ids = []
         fee = uint64(request.get("fee", 0))
         for nft_coin in request["nft_coin_list"]:
-            if "nft_coin_id" not in nft_coin or "nft_coin_id" not in nft_coin:
+            if "nft_coin_id" not in nft_coin or "wallet_id" not in nft_coin:
```
Comment (trepca) [inline chia/rpc/wallet_rpc_api.py:2219]: please add a test for this too (as in, old tests are passing with this bug)

**Review by trepca — APPROVED** [review]

**Pull request merged**